update.ps1

```diff
@@ -131,87 +136,65 @@ try {
+        Position            = $currentAccount.'urn:ietf:params:scim:schemas:extension:ysis:2.0:User'.position
+        MobilePhone         = ($($currentAccount.phoneNumbers) | Where-Object Type -eq 'mobile').value
+        WorkPhone           = ($($currentAccount.phoneNumbers) | Where-Object Type -eq 'work').value
+        UserName            = $currentAccount.userName
+        roles               = ($currentAccount.roles).displayname
+        exportTimelineEvents = $currentAccount.exportTimelineEvents
     }
 
-    # Set Username to existing (case-sensitive in Ysis)
-    if ($account.userName -ieq $currentAccount.userName) {        
-        $account.userName = $currentAccount.userName
-    }
-
-    # Set AGB to existing if null or empty
-    # if (!([string]::IsNullOrEmpty($previousAccount.AgbCode)) -and [string]::IsNullOrEmpty($account.AgbCode)) {
-    #     $account.AgbCode = $previousAccount.AgbCode
-    # }
-
-    # Set BIG to existing if null or empty
-    # if (!([string]::IsNullOrEmpty($previousAccount.BigNumber)) -and [string]::IsNullOrEmpty($account.BigNumber)) {
-    #     $account.BigNumber = $previousAccount.BigNumber
+    # Extra logic needs to be implemented when the username needs to be renamed when updating
+    # Set Username to existing if it already exists (case-sensitive in Ysis)
+    # if ($account.userName -ieq $currentAccount.userName) {
+    #     $account.userName = $currentAccount.userName
     # }
+    $account.userName = $currentAccount.userName # Do not rename the username
 
-    # Set MobilePhone to existing if null or empty
-    # if (!([string]::IsNullOrEmpty($previousAccount.MobilePhone)) -and [string]::IsNullOrEmpty($account.MobilePhone)) {
-    #     $account.MobilePhone = $previousAccount.MobilePhone
-    # }
-
-    # Set WorkPhone to existing if null or empty
-    # if (!([string]::IsNullOrEmpty($previousAccount.WorkPhone)) -and [string]::IsNullOrEmpty($account.WorkPhone)) {
-    #     $account.WorkPhone = $previousAccount.WorkPhone
-    # }
-
-    # Ysis account model mapping
-    # Roles are based on discipline and discipline can't be changed, so set Roles to existing    
-    # Modules could be changed manually, so set Modules to existing
-    # Discipline is immutable, so set Discipline to existing. When discipline is changed, notification will be send.
-
-    $ysisaccount = [PSCustomObject]@{
+    # Ysis account model mapping    # Discipline is immutable, so set Discipline to existing. When discipline is changed, notification will be send.
+    $ysisAccount = [PSCustomObject]@{
         schemas                                                      = @('urn:ietf:params:scim:schemas:core:2.0:User', 'urn:ietf:params:scim:schemas:extension:ysis:2.0:User', 'urn:ietf:params:scim:schemas:extension:enterprise:2.0:User')
         userName                                                     = $account.UserName
+        # userName                                                     = $currentAccount.UserName
```

Comment:
This line of code can be removed. Line 178-183 resolve this line

Reply:
updated